schema/constants.go

```diff
@@ -10,18 +10,23 @@ const (
 	CreateTableFollowingSQL                 = "create table following (peerID text primary key not null);"
 	CreateTableOfflineMessagesSQL           = "create table offlinemessages (url text primary key not null, timestamp integer, message blob);"
 	CreateTablePointersSQL                  = "create table pointers (pointerID text primary key not null, key text, address text, cancelID text, purpose integer, timestamp integer);"
-	CreateTableKeysSQL                      = "create table keys (scriptAddress text primary key not null, purpose integer, keyIndex integer, used integer, key text);"
-	CreateTableUnspentTransactionOutputsSQL = "create table utxos (outpoint text primary key not null, value integer, height integer, scriptPubKey text, watchOnly integer);"
-	CreateTableSpentTransactionOutputsSQL   = "create table stxos (outpoint text primary key not null, value integer, height integer, scriptPubKey text, watchOnly integer, spendHeight integer, spendTxid text);"
-	CreateTableTransactionsSQL              = "create table txns (txid text primary key not null, value integer, height integer, timestamp integer, watchOnly integer, tx blob);"
+	CreateTableKeysSQL                      = "create table keys (coin text, scriptAddress text primary key not null, purpose integer, keyIndex integer, used integer, key text);"
+	CreateIndexKeysSQL                      = "create index index_keys on keys (coin);"
+	CreateTableUnspentTransactionOutputsSQL = "create table utxos (coin text, outpoint text primary key not null, value integer, height integer, scriptPubKey text, watchOnly integer);"
+	CreateIndexUnspentTransactionOutputsSQL = "create index index_utxos on utxos (coin);"
+	CreateTableSpentTransactionOutputsSQL   = "create table stxos (coin text, outpoint text primary key not null, value integer, height integer, scriptPubKey text, watchOnly integer, spendHeight integer, spendTxid text);"
+	CreateIndexSpentTransactionOutputsSQL   = "create index index_stxos on stxos (coin);"
+	CreateTableTransactionsSQL              = "create table txns (coin text, txid text primary key not null, value integer, height integer, timestamp integer, watchOnly integer, tx blob);"
+	CreateIndexTransactionsSQL              = "create index index_txns on txns (coin);"
```

Comment:
I'm not sure if this is going to create any problems later on, but the column order in this schema is different from the column order that the migrations will produce. Which means that new nodes will have `coin` in the first column but migrated nodes will have `coin` in the last column. Could create a problem later on that would be very hard to detect and resolve.